### PR TITLE
[12.0][MIG] website_sale_vat_required: Migration to 12.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Supersedes https://github.com/OCA/e-commerce/pull/255

Cc @Tecnativa
TT19370

### CODE REVIEW

**Review by daramousk — APPROVED** [review]

**Comment by daramousk** [comment]
While this works great, it seems that the error on CI comes from `website_sale_require_login`. That seems to happen probably because https://github.com/OCA/e-commerce/blob/12.0/website_sale_require_login/static/src/js/website_sale_tour_buy_login.tour.js#L24

**Comment by rafaelbn** [comment]
@ernestotejeda please review travis :red_circle: 

`
2019-09-12 20:05:02,829 6558 INFO openerp_test odoo.addons.website_sale_vat_required.tests.test_website_sale_vat_required.TestUi: console log: - trigger 'div.o_has_error input[name='vat']': 1

2019-09-12 20:05:02,835 6558 INFO openerp_test odoo.addons.website_sale_vat_required.tests.test_website_sale_vat_required.TestUi: console log: - visible trigger 'div.o_has_error input[name='vat']': 0

2019-09-12 20:05:02,838 6558 INFO openerp_test odoo.addons.website_sale_vat_required.tests.test_website_sale_vat_required.TestUi: console log: - extra_trigger 'div.o_has_error input[name='vat']': 1

2019-09-12 20:05:02,840 6558 INFO openerp_test odoo.addons.website_sale_vat_required.tests.test_website_sale_vat_required.TestUi: console log: - visible extra_trigger 'div.o_has_error input[name='vat']': 0

2019-09-12 20:05:02,842 6558 INFO openerp_test odoo.addons.website_sale_vat_required.tests.test_website_sale_vat_required.TestUi: console log: _check_for_tooltip

2019-09-12 20:05:02,843 6558 INFO openerp_test odoo.addons.website_sale_vat_required.tests.test_website_sale_vat_required.TestUi: console log: - modal_displayed: 0

2019-09-12 20:05:02,845 6558 INFO openerp_test odoo.addons.website_sale_vat_required.tests.test_website_sale_vat_required.TestUi: console log: - trigger 'div.o_has_error input[name='vat']': 1

2019-09-12 20:05:02,847 6558 INFO openerp_test odoo.addons.website_sale_vat_required.tests.test_website_sale_vat_required.TestUi: console log: - visible trigger 'div.o_has_error input[name='vat']': 0

2019-09-12 20:05:02,848 6558 INFO openerp_test odoo.addons.website_sale_vat_required.tests.test_website_sale_vat_required.TestUi: console log: - extra_trigger 'div.o_has_error input[name='vat']': 1

2019-09-12 20:05:02,850 6558 INFO openerp_test odoo.addons.website_sale_vat_required.tests.test_website_sale_vat_required.TestUi: console log: - visible extra_trigger 'div.o_has_error input[name='vat']': 0
`

**Review by rafaelbn — APPROVED** [review]
:+1: 

Error is not related to this module, but please talk with @Yajo  and we can fix it @ernestotejeda 

Thanks

File: website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js [hunk 1]
```diff
@@ -1,40 +1,40 @@
 /* License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
  */
-
 odoo.define('website_sale_vat_required.tour', function(require) {
-
     "use strict";
 
-    require("website_sale.tour");
```
Comment (yajo) [inline website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js]: I didn't check if this changed in v13, but in general terms do not remove this. You are subject to a race condition without it.

File: website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
+            extra_trigger: "div.o_has_error input[name='vat']",
             run: function (actions) {
-                $('div.has-error input[name="vat"]').val("BE0477472701");
-                if ($('#div_phone').hasClass('has-error')){
+                $('div.o_has_error input[name="vat"]').val("BE0477472701");
+                if ($('#div_phone').hasClass('o_has_error')){
                     $('#div_phone input').val('11111111');
                 }
-                var input_accept_legal_terms = $('div.has-error input[name="accepted_legal_terms"]');
+                var input_accept_legal_terms = $('div input[name="accepted_legal_terms"]');
```
Comment (yajo) [inline website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js]: I'm not sure about this change. How are you checking that it has an error then?

**Review by yajo — CHANGES_REQUESTED** [review]
Code looks good at first sight, but there's something important removed.

**Comment by pedrobaeza** [comment]
Still failing

File: website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js (hunk 2 above)
Comment (ernestotejeda) [reply website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js]: I already changed it and left it as it was before

**Comment by ernestotejeda** [comment]
Ready to be reviewed

**Comment by pedrobaeza** [comment]
@Yajo can you confirm?

**Review by yajo — APPROVED** [review]

**Pull request merged**